### PR TITLE
Add options in support of web scraping tasks

### PR DESCRIPTION
**Pull request by fx2301**
I love the repo! Thanks for sharing it.

I added options to support my workflow for robust web scraping tasks. Feel free to take all / any that match your goals for this project. See README.md updates for documentation and reasoning.

### CODE REVIEW

File: har2requests/__init__.py [hunk 1]
```diff
@@ -188,6 +210,43 @@ def main(src, unsafe, no_infer, hide_result, include_options):
...
+                .replace("\n", "\n# "))
+
+        if generate_assertions:
+            expected_cookie_names = set(list(k for k,v in request.cookies.items() if v != '') + list(request.responseCookies.keys()))
+            for k in expected_cookie_names.copy():
+                if k in request.responseCookies and request.responseCookies[k] == '':
+                    expected_cookie_names.remove(k)
+            expected_cookie_names_safe = str(list(map(lambda str: re.sub("'", "\\'", str), expected_cookie_names)))
+            output(
+                f"assert set({expected_cookie_names_safe}) == set(s.cookies.get_dict().keys()), f\"Expected defined cookies to be {expected_cookie_names_safe} but was {{s.cookies.get_dict().keys()}} for url \\\"{request.url}\\\"\""
```
Comment (louisabraham) [inline har2requests/__init__.py:247]: wow that's some quote-fu

**Comment by louisabraham** [comment]
Nice, thanks for your contributions!

What do you think of also showing the returned data? It could be nice to be able to see what a typical answer looks like.

File: har2requests/__init__.py (hunk 1 above)
Comment (fx2301) [reply har2requests/__init__.py:247]: Ouch. Yes it is a bit over the top. Yuck.

**Comment by fx2301** [comment]
> Nice, thanks for your contributions!

You're welcome. This is a great tool!

> What do you think of also showing the returned data? It could be nice to be able to see what a typical answer looks like.

Tricky. Might involve some sniffing of the response content to see if it's binary and a heuristic around the size of the response? Check here for reference if you like: https://github.com/fx2301/mitmproxy-har-streamer/blob/master/stream.py#L75-L82

**Pull request merged**